### PR TITLE
Support tencent cloud COS storage in `datafusion-cli`

### PR DESCRIPTION
**Pull request by harveyyue**
## Which issue does this PR close?

NA

## What changes are included in this PR?
Support `cos://` schema as below:

```sql
CREATE EXTERNAL TABLE test
STORED AS PARQUET
OPTIONS(
    'aws.access_key_id' 'xxx',
    'aws.secret_access_key' 'xxx',
    'aws.cos.endpoint' 'https://cos.ap-singapore.myqcloud.com'
)
LOCATION 'cos://bucket/path/file.parquet';
```

## Are these changes tested?
yes, unit test

## Are there any user-facing changes?
no

### CODE REVIEW

File: docs/source/user-guide/cli.md [hunk 1]
```diff
@@ -380,6 +380,29 @@ The supported OPTIONS are:
...
+LOCATION 'cos://bucket/path/file.parquet';
+```
+
+The supported OPTIONS are:
+
+- access_key_id
+- secret_access_key
+- endpoint
+
+Note that the `endpoint` format of oss needs to be: `https://cos.{cos-region-endpoint}`
```
Comment (comphead) [inline docs/source/user-guide/cli.md:404]: cos instead of oss?

File: docs/source/user-guide/cli.md [hunk 2]
```diff
@@ -388,7 +411,7 @@ Note that the `endpoint` format of oss needs to be: `https://{bucket}.{oss-regio
 CREATE EXTERNAL TABLE test
 STORED AS PARQUET
 OPTIONS(
-    'service_account_path' '/tmp/gcs.json',
+    'gcp.service_account_path' '/tmp/gcs.json',
```
Comment (comphead) [inline docs/source/user-guide/cli.md:414]: is it related to cos onboard?

**Review by comphead — COMMENTED** [review]
Thanks @harveyyue for your contribution. 
Please add more details to the PR description, this will makes easier to other people what is the PR without going though the code.

Other than that it looks good, I'll leave it for other reviewers to have a second look.

File: datafusion-cli/src/object_storage.rs [hunk 3]
```diff
@@ -415,6 +424,15 @@ pub(crate) async fn get_object_store(
             let builder = get_oss_object_store_builder(url, options)?;
             Arc::new(builder.build()?)
         }
+        "cos" => {
+            let Some(options) = table_options.extensions.get::<AwsOptions>() else {
```
Comment (comphead) [inline datafusion-cli/src/object_storage.rs:428]: In follow up we need to refactor this part as for different schemas its too much of the common code

File: docs/source/user-guide/cli.md (hunk 1 above)
Comment (harveyyue) [reply docs/source/user-guide/cli.md:404]: No, just add cos external table doc.

File: docs/source/user-guide/cli.md (hunk 2 above)
Comment (harveyyue) [reply docs/source/user-guide/cli.md:414]: No, add the missing prefix gcp and aws to related external table options.

File: datafusion-cli/src/object_storage.rs (hunk 3 above)
Comment (harveyyue) [reply datafusion-cli/src/object_storage.rs:428]: Yes, agree with you.

File: docs/source/user-guide/cli.md (hunk 1 above)
Comment (alamb) [reply docs/source/user-guide/cli.md:404]:
```suggestion
Note that the `endpoint` format of urls must be: `https://cos.{cos-region-endpoint}`
```

File: datafusion-cli/src/object_storage.rs (hunk 3 above)
Comment (alamb) [reply datafusion-cli/src/object_storage.rs:428]: It would be great to file a ticket to do so 🙏

**Review by alamb — APPROVED** [review]
Looks good to me. Thank you @harveyyue  and @comphead  for the review

I took the liberty of fixing the clippy CI test and merging up from main and pushed the commits to your branch

**Comment by alamb** [comment]
Thanks again @harveyyue  and @comphead

**Pull request merged**